pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py

```diff
@@ -0,0 +1,1201 @@
+    ansible_repo_version_api_client,
+    build_and_upload_collection,
+    galaxy_v3_default_search_api_client,
+    gen_object_with_cleanup,
+):
+    """Make sure indexes are created for a repo with multiple distributions."""
+
+    build_config = {
+        "namespace": randstr(),
+        "name": randstr(),
+        "version": "1.0.1",
+    }
+
+    # make the repo
+    pulp_repo = gen_object_with_cleanup(ansible_repo_api_client, {"name": str(uuid.uuid4())})
+    repo_id = pulp_repo.pulp_href.split("/")[-2]
+
+    # add a collection to bump the version number
+    build_and_upload_collection(ansible_repo=pulp_repo, config=build_config)
+
+    # get the latest repo version ...
+    repository_versions = ansible_repo_version_api_client.list(pulp_repo.pulp_href)
+    repository_version_latest = repository_versions.results[0]
+
+    # make a distro that points at the first version
+    gen_object_with_cleanup(
+        ansible_distro_api_client,
+        {
+            "name": pulp_repo.name + "_v1",
+            "base_path": pulp_repo.name + "_v1",
+            "repository_version": repository_version_latest.pulp_href,
+        },
+    )
+
+    # we should now have 1 index
+    # repov1 + cv1
+
+    resp = galaxy_v3_default_search_api_client.list(limit=1000, repository=[repo_id])
+    assert resp.meta.count == 1
+    assert resp.data[0].repository_version == "1"
+
+    # add a collection to bump the version number
+    build_config["version"] = "1.0.2"
+    build_and_upload_collection(ansible_repo=pulp_repo, config=build_config)
+
+    # get the latest repo version ...
+    repository_versions = ansible_repo_version_api_client.list(pulp_repo.pulp_href)
+    repository_version_latest = repository_versions.results[0]
+
+    # make a distro that points at the second version
+    gen_object_with_cleanup(
+        ansible_distro_api_client,
+        {
+            "name": pulp_repo.name + "_v2",
+            "base_path": pulp_repo.name + "_v2",
+            "repository_version": repository_version_latest.pulp_href,
+        },
+    )
+
+    # per david, we should have 3 entries now
+    # repov1+cv1
+    # repov2+cv1
+    # repov2+cv2
+
+    resp = galaxy_v3_default_search_api_client.list(limit=1000, repository=[repo_id])
+    assert resp.meta.count == 3
+    repo_versions = sorted([int(x.repository_version) for x in resp.data])
+    assert repo_versions == [1, 2, 2]
+
+    # add a collection to bump the version number
+    build_config["version"] = "1.0.3"
+    build_and_upload_collection(ansible_repo=pulp_repo, config=build_config)
+
+    # make a distro that points at the repository instead of a version
+    gen_object_with_cleanup(
+        ansible_distro_api_client,
+        {
+            "name": pulp_repo.name + "_vN",
+            "base_path": pulp_repo.name + "_vN",
+            "repository": pulp_repo.pulp_href,
+        },
+    )
+
+    # per david, we should have 6 entries now
+    # v1,2,3 w/ rv=null, v1,2 with rv=1,2, v1 with rv=1
```

Comment:
```suggestion
    # cv1,2,3 w/ rv=null, cv1,2 with rv=1,2, cv1 with rv=1
```
I think this is what you meant, the comment is confusing otherwise.